backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/api/IConnectorToolsProvider.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.collaborative.diagrams.api;
+
+import java.util.List;
+
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.diagrams.description.DiagramDescription;
+import org.eclipse.sirius.components.diagrams.tools.ITool;
+
+/**
+ * Provides tools for the connector tool.
+ *
+ * @author nvannier
+ */
+public interface IConnectorToolsProvider {
+
+    boolean canHandle(DiagramDescription diagramDescription);
+
+    List<ITool> getConnectorTools(Object sourceDiagramElement, Object targetDiagramElement, IObjectService objectService, IEditingContext editingContext);
```

Comment:
Do not provide the object service here. It's a service class which can be injected by those who will contribute an implementation of this class.